### PR TITLE
Completed baseline section with basic responsiveness

### PR DESCRIPTION
**Pull request by alvinleung**
Here is the high level overview on what's inside this gigantic update. Sorry I didn't have time to write all the docs because the amount of code we have here and with the interest of trading velocity with quality. But I can setup a call to walk you guys through it if you are interested in using certain part of it.

- With fully refactored carousel components 
- Added `TracyShdow` component for the glow effect
- Added `AnimationConfig` file for single source of truth for all the motion easing
- Added `useComponentControls` for exposing internal component functionality to its parent component (think framer motions useDragControls), for its usage, please reference how it was used in the carousel.
- Added `useFollowMotionValue`—use it when you want a `MotionValue` trail the value of another `MotionValue`

### CODE REVIEW

**Comment by alvinleung** [comment]
Fixed the mobile scrolling bug, turns out it is about how the browser handle pointer move. [For future reference](https://stackoverflow.com/questions/48124372/pointermove-event-not-working-with-touch-why-not)

File: components/Carousel/Carousel.tsx [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import {
+  ComponentControls,
+  useComponentControls,
+} from "@/hooks/useComponentControls";
+import useStateRef from "@/hooks/useStateRef";
```
Comment (christopher-ha) [inline components/Carousel/Carousel.tsx:25]: Where is this being used?

Comment (alvinleung) [reply components/Carousel/Carousel.tsx:25]: It is used to define the `useCarouselControl` down there

**Review by christopher-ha — APPROVED** [review]
Bug is fixed and component works as intended.

Minor: Create TracyShadow.tsx as folder and inside create index.tsx for the component logic. That way every component is its own folder and we dont have a mix of folders and files. 

For next push, let's try to find a place for the AnimationConfig.ts and potentially TypeSystemPlugin.ts

File: components/Carousel/Carousel.tsx (hunk 1 above)
Comment (christopher-ha) [reply components/Carousel/Carousel.tsx:25]: Sorry, meant the useStateRef.

Comment (christopher-ha) [reply components/Carousel/Carousel.tsx:25]: In the component it's being imported but not used anywhere. Could potentially remove this if it's not used for the component to reduce logic overhead.

**Pull request merged**